### ISSUE TITLE
Strange freezing up every 5-10 minutes.

### ISSUE THREAD

**Issue by chariegg** [trelby/trelby#19]
Every 5 or 10 minutes or so I can't input any more text or click on anything anymore forcing to press Ctrl + C if opened via terminal or change tty and terminate python. This attached log is all I could get out of it trying to replicate it, it was very unpredictable how and when it happens. I am on debian 12, and installed all the deps via pip/venv and used the git branch rather than the tarball release from 2017.
[trelbyerrors.txt](https://github.com/trelby/trelby/files/14027668/trelbyerrors.txt)

Only changes I made to this document was changing in another text editor the full path with $HOME.

**Comment by limburgher** [comment]
Does using the deb version from OBS work better?

https://build.opensuse.org/package/show/home:gwync/trelby

**Comment by chariegg** [comment]
> Does using the deb version from OBS work better?
> 
> https://build.opensuse.org/package/show/home:gwync/trelby

No, but I did find at least one way to replicate it. Hold enter and make a few pages. You know how you can press ctrl + up or ctrl + down to jump to the start or end. Now press ctrl + left or ctrl + right. No more typing but you can still use the mouse. This is exactly like how it was like (if it didn't freeze the mouse also) as previously. I'm on GNOME x11, if that makes a difference.

**Comment by chariegg** [comment]
@limburgher 
What is also strange how the older python2 based (version 2.2 from the downloads) works better both on windows and on linux via wine. I can scroll properly, and so far (tested for a few minutes) it hasn't been able to reproduce issues. This should still be investigated, just wanted to add if anyone else is having problems, give the older one a try.

**Comment by chariegg** [comment]
@limburgher 
This issue of keyboard freezing simply doesn't happen on windows. Only on linux. I want to just say that it's a GTK bug because wxwidgets uses GTK as the actual GUI, but I don't want to jump the gun on this.

**Comment by limburgher** [comment]
Interesting. I can't reproduce this with Python 3 on Fedora. What versions of wxwidgets, gtk3, etc, do you have? Do you see it under another desktop environment, or using Wayland?

**Comment by chariegg** [comment]
@limburgher 
The versions for GTK3, wx, and wxpython on debian 12 (stable) are:
python3-wxgtk4.0 is at 4.2.0
libwxgtk3.2-1 is at 3.2.2
libgtk-3-0 is at 3.22.25
python3 is 3.11.12


Fedora on the other hand:
python3 is at 3.12
wxGTK is at 3.2.2.1
gtk3 is 3.24.38-3

So is GTK 3.22 not enough? It must be at 3.24? Just to make sure, what DE/WM for Fedora did you test this on? I tested this on debian with GNOME and even something super minimalist like JWM. Both got the same results. I think debian for one reason or another, having a slightly older version is making this difficult with trelby. What do you think?

**Comment by chariegg** [comment]
@limburgher 
It doesn't work for me on fedora 39 with GNOME. I haven't tested it with KDE yet, but it hates my setup on two different computers, with two different distros. I can still use the windows version via wine at least.

**Comment by chariegg** [comment]
@limburgher 
I for the life of me can't figure out what is going on. I can still do some modifier keys like ctrl + f, ctrl + s, or crl +q even if I can't type any more regular keyboard characters like asdfjkl after getting no more typing after doing stuff like ctrl + left or anything else that causes it. I figured out two things by chance: Quitting the program fixes it, but opening or switching tabs and typing a bit and then returning fixes it also. I guess I found a workaround, but if anyone else comes across is, please keep it open until this is permanently fixed. Because I give up on figuring out why. Nothing, gdb or pdb, or python -vvvv or journalctl or the regular terminal stdout will report anything happening.

**Comment by alexlnkp** [comment]
> > Does using the deb version from OBS work better?
> > [build.opensuse.org/package/show/home:gwync/trelby](https://build.opensuse.org/package/show/home:gwync/trelby)
> 
> No, but I did find at least one way to replicate it. Hold enter and make a few pages. You know how you can press ctrl + up or ctrl + down to jump to the start or end. Now press ctrl + left or ctrl + right. No more typing but you can still use the mouse. This is exactly like how it was like (if it didn't freeze the mouse also) as previously. I'm on GNOME x11, if that makes a difference.

Can confirm, same thing happens to me if I try to use Ctrl+Left or Ctrl+Right. Keyboard starts just being ignored. Ctrl+F and Ctrl+S still work as shortcuts, though.
No output printed to the terminal when that occurs.

version: `trelby 2.4.10-1`
from AUR, installed via `yay -S trelby`

specs:
```neofetch
OS: Arch Linux x86_64
Kernel: Linux 6.12.4-arch1-1
WM: Hyprland (Wayland)
CPU: 12th Gen Intel(R) Core(TM) i5-12400F (12) @ 4.40 GHz
GPU: NVIDIA GeForce RTX 3060 Lite Hash Rate [Discrete]
Memory: 6.59 GiB / 31.20 GiB (21%)
```

**Comment by Ian2020** [comment]
I get this with the flatpak also.

**Comment by librafrog** [comment]
Think it could be a glitch in windows as well - v.2.4.16.2 PyPi - pip. I try to limit system keyboard shortcuts involving <kbd>Ctrl</kbd>
